python/main.py

```diff
@@ -62,20 +60,26 @@ def root():
 
 
 @app.post("/items")
-def add_item(name: str = Form(...), category: str = Form(...)):
+async def add_item(name: str = Form(...), category: str = Form(...), image: UploadFile = File(...)):
     logger.info(f"Receive item - name:{name}, category:{category}")
 
+    if not image.filename.endswith(".jpg"):
+        raise HTTPException(
+            status_code=400, detail="Image is not in .jpg format")
+
+    hashes = hashlib.sha256(
+        image.filename.split(".")[0].encode('utf-8')).hexdigest() + '.jpg'
+
```

Comment:
Thank you for updating!

It will still work as long as the server is running.
However, the files will be lost when the server or PC is restarted.

I guess you need to store `image` in the disk.

https://fastapi.tiangolo.com/tutorial/request-files/#file-parameters-with-uploadfile

> file: A [SpooledTemporaryFile](https://docs.python.org/3/library/tempfile.html#tempfile.SpooledTemporaryFile)

Reply:
Thank you for advice! I added some line to main.py. Without adding those lines, I wasn't able to retrieve the image at Step 5, so yes it is very necessary step. Thank you!